Core: Add `preview.js` default export support

kasperpeulen opened the pull request:
Make sure that users can use export default in a `preview.ts/js` file as well, as alternative of named exports.

yannbf committed:
provide packageManager when copying template files

kasperpeulen committed:
Try to fix it instead

yannbf commented on code/lib/preview-api/src/modules/store/csf/composeConfigs.test.ts:
```diff
@@ -42,20 +42,52 @@ describe('composeConfigs', () => {
-          argTypes: { x: '2', z: '2', obj: { a: '2', c: '2' } },
-          globals: { x: '2', z: '2', obj: { a: '2', c: '2' } },
-          globalTypes: { x: '2', z: '2', obj: { a: '2', c: '2' } },
+          default: {
```
Could you add a case where this is mixed? We are composing configs that could come from users, addons, presets and frameworks, which will definitely be mixed and the composing of it all should work correctly

shilman replied:
Will do once CI finishes & merge

yannbf replied:
It's all good – I tested that and added a telescoping PR for after we merge this: https://github.com/storybookjs/storybook/pull/21230

yannbf committed:
add mixed exports test case

yannbf committed:
Merge pull request #21230 from storybookjs/test/add-additional-composeconfig-test-case

Build: Add mixed exports test case to composeConfigs

kasperpeulen committed:
Fix default export for preview in ssv6

kasperpeulen committed:
Merge remote-tracking branch 'origin/rollback-preview-ts-template' into rollback-preview-ts-template

kasperpeulen committed:
Fix default export for preview in ssv6 (for real)

kasperpeulen committed:
Merge remote-tracking branch 'origin/next' into rollback-preview-ts-template

tmeasday committed:
Only skip yarn setup for PNP

This was breaking the vue-cli templates and seems to be purely for pnp purposes:

https://github.com/storybookjs/storybook/pull/21046/files#diff-76aad599d2311ab4fec53db693f2e85d4cb7f798cc37357dfe7839997caef2ba

tmeasday committed:
Remove vue-cli as sandbox template is currently busted
